scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/synchronization/AOSynchronization.java

```diff
@@ -0,0 +1,1038 @@
+        if (!statusFileDirectory.exists()) {
+            boolean result = statusFileDirectory.mkdirs();
+            if (!result) {
+                throw new IllegalArgumentException("Could not create directory hierarchy for path " +
+                                                   statusFileDirectory);
+            }
+        } else if (!statusFileDirectory.isDirectory()) {
+            throw new IllegalArgumentException("Provided directory path exists and is not a directory " +
+                                               statusFileDirectory);
+        }
+        this.statusFile = new File(statusFileDirectory, DEFAULT_STORE_SESSION_NAME);
+    }
+
+    /**
+     * Initialize the Record Manager.
+     *
+     * Load the persisted channel from disk at configured location.
+     *
+     * Delete the persisted database in case a serious error occurred when loading the database.
+     * @param firstAttempt first attempt at loading the persisted channels
+     * @throws IllegalStateException when an error occurred during database loading, even after cleaning it
+     */
+    private void init(boolean firstAttempt) {
+        close();
+        try {
+            logger.info("Loading Record Manager from file : " + statusFile);
+            recordManager = RecordManagerFactory.createRecordManager(statusFile.getCanonicalPath());
+            persistedChannels = recordManager.hashMap(STATUS_RECORD_NAME);
+            // While printing the content of persistedChannels, InvalidClassException or jdbm Error can be triggered
+            logger.info("Content of persisted store : " + persistedChannels);
+
+            recordManager.commit();
+            if (!firstAttempt) {
+                logger.info("Loading of job database successful after clean.");
+            }
+        } catch (Throwable e) {
+            if (firstAttempt) {
+                logger.error("Error occurred when loading channels database " + statusFile.getAbsolutePath() +
+                             ", now cleaning it and retrying.", e);
+                cleanDataBase();
+                init(false);
+            } else {
+                close();
+                throw new IllegalStateException("Error when loading database (even after cleaning it): " +
+                                                statusFile.getAbsolutePath(), e);
+            }
+        }
+        isStarted = true;
+    }
+
+    public boolean isStarted() {
+        return isStarted;
+    }
+
+    private void cleanDataBase() {
+        close();
+        logger.info("Cleaning database");
+
+        // delete all db files
+        File[] dbChannelFiles = statusFileDirectory.listFiles((dir,
+                name) -> name.startsWith(DEFAULT_STORE_SESSION_NAME));
+        if (dbChannelFiles != null) {
+            for (File file : dbChannelFiles) {
+                try {
+                    logger.info("Deleting " + file);
+                    boolean result = file.delete();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Use "java.nio.Files#delete" here for better messages on error conditions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4042)